### ISSUE TITLE
Update withGluestack.ts with options parameter, like withNativebase.ts

### ISSUE THREAD

**Issue by haveamission** [gluestack/ui-next-adapter#10]
Right now, withGluestack.ts does not include an options parameter on webpack, this means that you aren't allowed to use any options in your custom webpacks, which gives less customization/flexibility.

I propose changing this line (withGluestack.ts, line 100):

`webpack: (config: any) => {`

to this line:

`webpack: (config: any, options: any) => {`

This is more consistent with NextJS documentation as well.

**Comment by Gwened** [comment]
Indeed, the NextJS documentation includes a second argument which has the following structure:
`{ buildId, dev, isServer, defaultLoaders, nextRuntime, webpack }`
See https://nextjs.org/docs/app/api-reference/next-config-js/webpack

As a workaround, one can move their webpack section out of the passed nextConfig object and do something like that:
```
const configWithGluestack = withGluestackUI(nextConfig);
export default {
    ...configWithGluestack,
    webpack: (
        config,
        { buildId, dev, isServer, defaultLoaders, nextRuntime, webpack }
      ) => {
        //  Do what you need to do here
        // ...
        return configWithGluestack.webpack(config);
      }
}
```
